src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -110,38 +116,72 @@ void DeferredCredits::apply(
             to.map_.emplace(p);
         if (! r.second)
         {
-            using std::get;
-            get<0>(r.first->second) += get<0>(p.second);
-            get<1>(r.first->second) += get<1>(p.second);
+            auto& toVal = r.first->second;
+            auto const& fromVal = p.second;
+            toVal.lowAcctCredits += fromVal.lowAcctCredits;
+            toVal.highAcctCredits += fromVal.highAcctCredits;
+            // Do not update the orig balance, it's already correct
         }
     }
 }
 
-void DeferredCredits::clear ()
-{
-    map_.clear ();
-}
-
 } // detail
 
 STAmount
 PaymentSandbox::balanceHook (AccountID const& account,
     AccountID const& issuer,
         STAmount const& amount) const
 {
-    if (ps_)
-        return tab_.adjustedBalance (
-            account, issuer, ps_->balanceHook (account, issuer, amount));
-    return tab_.adjustedBalance(
-        account, issuer, amount);
+    /*
+    There are two algorithms here. The first takes the current amount and
+    subtracts the recorded credits. The second algorithm remembers the original
+    balance, and subtracts the debits. The second algorithm should be more
+    numerically stable than the first. Consider a large credit with a small
+    initial balance. The first algorithm computes (B+C)-C (where B+C will the
+    the amount passed in). The second algorithm return B. When B and C differ by
+    large magnitudes, (B+C)-C may not equal B.
+    */
```

Comment:
This is a very useful paragraph.  Possibly, for extra clarity, you might refer to the algorithms as pre-switchover and post-switchover?  As it is I believe the "first" algorithm comes later in the flow of the code than the "second" algorithm.